tests/TypeTokenTest.php

```diff
@@ -200,11 +200,10 @@ public function testNestedGeneric()
 
     public function testGenericNoEndingBracket()
     {
-        try {
-            new TypeToken('array<string');
-        } catch (MalformedTypeException $exception) {
-            self::assertSame('Could not find ending ">" for generic type', $exception->getMessage());
-        }
+        self::expectException(MalformedTypeException::class);
+        self::expectExceptionMessage('Could not find ending ">" for generic type');
```

Comment:
The reason this is wrapped in a try/catch is because `expectExceptionMessage` matches on substrings instead of the full string. This causes tests to pass when you may want them to fail. The best way to handle this is the try/catch.

Reply:
In this test, the exception message is the `Could not find ending ">" for generic type`.
The `try...catch` approach or `expectException` approach get the same result.
The test is passed because the test expects the exception and test will be passed.
The proper way is the `expectException` approach I think :+1: .

Reply:
But if the exception message was longer, or the expected message was shorter, the test would still pass.

Reply:
If possible, can you show some test cases for this?

Reply:
https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/Constraint/ExceptionMessage.php#L47

Reply:
I know that, but this will not happen in this test because these two approaches are the same expected exception messages.

Reply:
Until the exception message changes and the test does not fail to reflect that change.

Reply:
It can be difficult to track down every place an exception is tested in a bigger library. Additionally, since the exception messages are something that's being tested, I expected them to fail when changes are made. I don't want to have to remember to find every test that's testing that exception. I've found that this is the best way to handle it, and it's consistent across the libraries I write.

Reply:
@natebrunette, I know that. I will not hand on this because of the consistency.
I've reverted this.